python/hf-storage-initializer.Dockerfile

```diff
@@ -0,0 +1,52 @@
+ARG PYTHON_VERSION=3.9
+ARG BASE_IMAGE=python:${PYTHON_VERSION}-slim-bullseye
+ARG VENV_PATH=/prod_venv
+
+FROM ${BASE_IMAGE} as builder
+
+# Install Poetry
+ARG POETRY_HOME=/opt/poetry
+ARG POETRY_VERSION=1.7.1
+
+# Required for building packages for arm64 arch
+RUN apt-get update && apt-get install -y --no-install-recommends python3-dev build-essential && apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN python3 -m venv ${POETRY_HOME} && ${POETRY_HOME}/bin/pip install poetry==${POETRY_VERSION}
+ENV PATH="$PATH:${POETRY_HOME}/bin"
+
+# Activate virtual env
+ARG VENV_PATH
+ENV VIRTUAL_ENV=${VENV_PATH}
+RUN python3 -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+
+COPY kserve/pyproject.toml kserve/poetry.lock kserve/
+RUN cd kserve && poetry install --no-root --no-interaction --no-cache --extras "hf-storage"
+COPY kserve kserve
```

Comment:
does storage initializer actually need the kserve dependency?

Reply:
a question, can't the existing storage initializer image be reused?
Thinking that users would need to switch manually the initializer images

Reply:
is it possible to reduce the image size by using some other light-weighted sdk? adding a new image means the operation team needs to maintain a new image(security scan etc) just for models from HF.

Reply:
okay, thanks.

Reply:
btw, I am working on this https://github.com/kserve/kserve/issues/3489
it is related to it.

the plan is to decouple the storage package from KServe and make a hard copy of the logging.py file, so the storage-initializer can be completely independent from KServe avoiding importing/installing a few unneeded dependencies, like ray and others.
Like